core/test-extension/deployment/src/test/java/io/quarkus/naming/DisableJNDITestCase.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.naming;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Hashtable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class DisableJNDITestCase {
+
+    @RegisterExtension
+    static QuarkusUnitTest unitTest = new QuarkusUnitTest();
+
+    @Test
+    public void testJNDIDisabled() throws Exception {
+        CompletableFuture<Boolean> cf = new CompletableFuture<>();
+        //we just want to test that there is a connection attempt
+        //we open a socket and will fail the test if anything actually manages
+        //to connect to it
+        try (ServerSocket s = new ServerSocket(1389)) {
```

Comment:
Not sure how useful this assertion is. AFAIU this test will always fail if an LDAP server (or anything else bound to this port) is running locally.

Reply:
Perhaps let it bind to a random port and then use this port instead of `1389` in the subsequent assertions